[ERROR] Unknown Member DiscordAPIError[10007]: Unknown Member

We should add more logging to know to which thread/post this happens to investigate why this issue happens.

**Acceptance criteria:**
- The error should be more descriptive, includes thread/post and possible user name.

Error stack trace:

`[ERROR] Unknown Member DiscordAPIError[10007]: Unknown Member
    at handleErrors (/build/reactibot/node_modules/@discordjs/rest/dist/index.js:730:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async SequentialHandler.runRequest (/build/reactibot/node_modules/@discordjs/rest/dist/index.js:1133:23)
    at async SequentialHandler.queueRequest (/build/reactibot/node_modules/@discordjs/rest/dist/index.js:963:14)
    at async _REST.request (/build/reactibot/node_modules/@discordjs/rest/dist/index.js:1278:22)
    at async GuildMemberManager._fetchSingle (/build/reactibot/node_modules/discord.js/src/managers/GuildMemberManager.js:221:18)
    at async Promise.all (index 13)
    at async Promise.all (index 0)
    at async Object.handleReaction (file:///build/reactibot/dist/features/emojiMod.js:103:49)
`

--- a/src/features/emojiMod.ts
+++ b/src/features/emojiMod.ts
@@ -21,6 +21,7 @@ import {
 } from "../helpers/discord.js";
 import { partition } from "../helpers/array.js";
 import { EMBED_COLOR } from "./commands.js";
+import { logger } from "./log.js";
 
 const config = {
   // This is how many ️️warning reactions a post must get until it's considered an official warning
@@ -146,31 +147,45 @@ const emojiMod: ChannelHandlers = {
     if (!reactionHandlers[emoji]) {
       return;
     }
+    try {
+      const [fullReaction, fullMessage, reactor] = await Promise.all([
+        reaction.partial ? reaction.fetch() : reaction,
+        message.partial ? message.fetch() : message,
+        guild.members.fetch(user.id),
+      ]);
+      const [usersWhoReacted, authorMember] = await Promise.all([
+        fetchReactionMembers(guild, fullReaction),
+        guild.members.fetch(fullMessage.author.id),
+      ]);
 
-    const [fullReaction, fullMessage, reactor] = await Promise.all([
-      reaction.partial ? reaction.fetch() : reaction,
-      message.partial ? message.fetch() : message,
-      guild.members.fetch(user.id),
-    ]);
-    const [usersWhoReacted, authorMember] = await Promise.all([
-      fetchReactionMembers(guild, fullReaction),
-      guild.members.fetch(fullMessage.author.id),
-    ]);
+      if (authorMember.id === bot.user?.id) return;
 
-    if (authorMember.id === bot.user?.id) return;
+      if (authorMember.id === bot.user?.id) return;
 
-    if (authorMember.id === bot.user?.id) return;
-
-    reactionHandlers[emoji]({
-      guild,
-      author: authorMember,
-      reactor,
-      message: fullMessage,
-      reaction: fullReaction,
-      usersWhoReacted: usersWhoReacted.filter(
-        (x): x is GuildMember => Boolean(x) && authorMember.id !== reactor.id,
-      ),
-    });
+      reactionHandlers[emoji]({
+        guild,
+        author: authorMember,
+        reactor,
+        message: fullMessage,
+        reaction: fullReaction,
+        usersWhoReacted: usersWhoReacted.filter(
+          (x): x is GuildMember => Boolean(x) && authorMember.id !== reactor.id,
+        ),
+      });
+    } catch (e) {
+      if (e instanceof Error) {
+        let descriptiveMessage = `Channel id: ${message.channelId}`;
+        if (message.channel.isThread()) {
+          const thread = message.channel;
+          descriptiveMessage += `Thread id: ${thread.id}`;
+        }
+        logger.log(
+          `${descriptiveMessage} username: ${user.username}  message: ${message.content}`,
+          e,
+        );
+      }
+      throw e;
+    }
   },
 };
 

